TestAssets/TestProjects/AppWithDependencyOnMalformedPackage/project.json

```diff
@@ -0,0 +1,14 @@
+{
+  "version": "1.0.0-*",
+  "compilationOptions": {
+    "emitEntryPoint": true
+  },
+  "dependencies": {
+    "BadPackage": "1.0.0"
```

Comment:
It's hard to generate, because the bad dll should be present in lib but not in ref.

Right now I'd vote for deleting the test and its test data, and just keeping the exception handling. It's more trouble than it's worth :)